### PR TITLE
Fix gradient not being recorded

### PR DESCRIPTION
**Pull request by gmuraru**
## Description
Track also the operations from static using the `forward` method call.
We tracked only the operations we did on ```MPCTensor```, but we need to be able to also track the operations from ```functional```.

## Affected Dependencies
The `relu` and `maxpool_2d` grad.

## How has this been tested?
- The tests should pass as they are

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: src/sympc/tensor/mpc_tensor.py [hunk 1]
```diff
@@ -865,21 +865,21 @@ def __getattribute__(self, attr_name: str) -> Any:
...
         session = object.__getattribute__(self, "session")
         if grad_fn and session.autograd_active:
-            from sympc.tensor.grads import forward
+            from sympc.grads import forward
 
             return functools.partial(forward, self, grad_fn)
 
+        if attr_name in STATIC_FUNCS.keys():
+            return functools.partial(STATIC_FUNCS[attr_name], self)
+
```
Comment (Syzygianinfern0) [inline src/sympc/tensor/mpc_tensor.py:882]: Why this move?

File: src/sympc/grads/grad_functions.py [hunk 2]
```diff
@@ -636,6 +636,7 @@ def forward(
         stride: Optional[Union[int, Tuple[int, int]]] = None,
         padding: Union[int, Tuple[int, int]] = 0,
         dilation: Union[int, Tuple[int, int]] = 1,
+        return_indices: bool = False,
```
Comment (Syzygianinfern0) [inline src/sympc/grads/grad_functions.py:639]: Missed out forwarding this arg to the call below (L679)

https://github.com/OpenMined/SyMPC/blob/969718db81690b674a642b29984074fd45286034/src/sympc/grads/grad_functions.py#L673-L680

**Review by Syzygianinfern0 — CHANGES_REQUESTED** [review]

File: src/sympc/tensor/mpc_tensor.py (hunk 1 above)
Comment (gmuraru) [reply src/sympc/tensor/mpc_tensor.py:882]: I was thinking that the Grad should be checked first and then the statics --- only like an "importance hierarchy" -- for the moment I do not think we have any grad recording in the static file.

File: src/sympc/grads/grad_functions.py (hunk 2 above)
Comment (gmuraru) [reply src/sympc/grads/grad_functions.py:639]: I should have had an if...else statement - the return indices would be needed if we record the gradient.

File: tests/sympc/grad/grad_functions_test.py [hunk 3]
```diff
@@ -7,23 +7,23 @@
 import torch
 
 import sympc
+from sympc.grads.grad_functions import GradAdd
```
Comment (gmuraru) [inline tests/sympc/grad/grad_functions_test.py:10]: This was done by ```isort```

File: src/sympc/module/nn/functional.py [hunk 4]
```diff
@@ -24,6 +26,9 @@ def relu(x: MPCTensor) -> MPCTensor:
     Returns:
         An MPCTensor which represents the ReLu applied on the input tensor
     """
+    relu_forward = GRAD_FUNCS.get("relu", None)
+    if relu_forward and x.session.autograd_active:
+        return forward(x, relu_forward)
```
Comment (aanurraj) [inline src/sympc/module/nn/functional.py:31]: this will the issue for not recording the grad right? but I wonder where is it being stored as node ?

**Review by aanurraj — APPROVED** [review]
Thank you, just one more question have you ran the notebook which I gave? it is working on that architecture?

File: src/sympc/module/nn/functional.py (hunk 4 above)
Comment (gmuraru) [reply src/sympc/module/nn/functional.py:31]: It is recorded - the forward function deals with this.

**Comment by gmuraru** [comment]
> Thank you, just one more question have you ran the notebook which I gave? it is working on that architecture?

I tested it on the code you gave me on slack - but thank you for remembering me that I did not uncomment some of the lines.

**Comment by gmuraru** [comment]
> Thank you, just one more question have you ran the notebook which I gave? it is working on that architecture?

Good call @aanurraj. I have found a problem with the `reshape` and I fixed it ^_^
